clients/drcachesim/tools/view.cpp

```diff
@@ -205,6 +157,78 @@ view_t::process_memref(const memref_t &memref)
     return parallel_shard_memref(serial_stream_, memref);
 }
 
+bool
+view_t::init_decode_cache()
+{
+    assert(decode_cache_ == nullptr);
+    decode_cache_ =
+        std::unique_ptr<decode_cache_t<disasm_info_t>>(new decode_cache_t<disasm_info_t>(
+            dcontext_.dcontext,
+            // We will decode the instr_t ourselves.
+            /*include_decoded_instr=*/false,
+            /*persist_decoded_instrs=*/false, knob_verbose_));
+    if (TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, filetype_)) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_));
+    } else if (!module_file_path_.empty()) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_),
+                                            module_file_path_, knob_alt_module_dir_);
+    } else {
+        // Continue but omit disassembly to support cases where binaries are
+        // not available and OFFLINE_FILE_TYPE_ENCODINGS is not present.
+        decode_cache_.reset(nullptr);
+    }
+    return error_string_.empty();
+}
+
+bool
+view_t::init_from_filetype()
+{
+    if (init_from_filetype_done_) {
+        return true;
+    }
+    if (filetype_ == OFFLINE_FILE_TYPE_DEFAULT) {
+        filetype_ = static_cast<offline_file_type_t>(serial_stream_->get_filetype());
+    }
+    assert(filetype_ != OFFLINE_FILE_TYPE_DEFAULT);
+    if (!init_decode_cache()) {
+        return false;
+    }
+    // We remove OFFLINE_FILE_TYPE_ARCH_REGDEPS from this check since
+    // DR_ISA_REGDEPS is not a real ISA and can coexist with any real
+    // architecture.
+    if (TESTANY(OFFLINE_FILE_TYPE_ARCH_ALL & ~OFFLINE_FILE_TYPE_ARCH_REGDEPS,
+                filetype_) &&
```

Comment:
Doesn't this check contradict your assertion above that _DEFAULT is an invalid type?  This check implies a trace doesn't have to have any arch flag set.

Reply:
You mean the fact that the first part of the conjunction even exists hints that we expected there to be traces without any arch bit set?

Reply:
Reverted for now. See other comment thread.